sdk/scheduler/src/main/java/com/mesosphere/sdk/api/StateResource.java

```diff
@@ -170,6 +168,59 @@ public Response putFile(
         }
     }
 
+    /**
+     * Returns the Zone information for all of the tasks of the service.
+     */
+    @Path("/zone/tasks")
+    @GET
+    public Response getTaskNamesToZones() {
```

Comment:
Do all of these APIs get used? I don't see the `zone-resolver.sh` script anywhere in this PR?

Reply:
Nope, only the `getTaskIPsToZones` handler. Nick suggested we might as well expose a couple other endpoints to get zone info from, but these can be removed if you have a strong preference (I don't.)

Reply:
👍

Reply:
Are there tests for the other *Resource files? If yes then...

Reply:
Yup, there are now